saleor/csv/tasks.py

```diff
@@ -59,3 +68,30 @@ def export_gift_cards_task(
 ):
     export_file = ExportFile.objects.get(pk=export_file_id)
     export_gift_cards(export_file, scope, file_type, delimiter)
+
+
+@app.task
+def delete_old_export_files():
+    now = timezone.now()
+
+    events = ExportEvent.objects.filter(
+        type=ExportEvents.EXPORT_PENDING,
+        date__lte=now - settings.EXPORT_FILES_TIMEDELTA,
+    ).values("export_file_id")
+    export_files = ExportFile.objects.filter(
+        Q(events__isnull=True) | Exists(events.filter(export_file_id=OuterRef("id")))
+    )
+
+    if not export_files:
+        return
```

Comment:
Shouldn't queryset require `exists ()` to check for empty?

Reply:
No, it'll work too :)